rust/kernel/task.rs

```diff
@@ -96,6 +96,12 @@ impl Task {
         unsafe { (*self.ptr).pid }
     }
 
+    /// Returns the Mm of the given task.
+    pub fn mm(&self) -> &MmStruct {
+        let ptr = unsafe { core::ptr::addr_of!((*self.ptr).mm).read() };
```

Comment:
Why did you use `addr_of!()` and `.read()` here instead of `(*self.ptr).mm`?

Reply:
I'm not very sure if there are any semantics for read_once() using adds_of!() and .read().

Reply:
I think the closest to `read_once` is `read_volatile` in rust. @ojeda?

Reply:
> It depends on the architecture/config, e.g. in arm64 with LTO `READ_ONCE` gets acquire semantics. @wedsonaf?

I believe this style was copied from `File`, where I have a [comment](https://github.com/Rust-for-Linux/linux/blob/rust/rust/kernel/file.rs#L32-L34) lamenting the situation. My thinking of using `addr_of` + `read` there was to reduce the window of potential UB.

Reply:
"(*self.ptr).mm " might be enough.